### PR TITLE
allow separate key binding for the numerics keypad

### LINKED ISSUE #1147: allow to bind keypad keys separately
It seems that at least USB keyboards send separate scancodes for the keypad buttons. We should probably allow to bind them separately from their primary or secondary function. 

Or is there any disadvantage that I just can't see at the moment? 

While at it, should we allow this for the numbers only or also for the plus, minus, asterisk, slash keys as well? The keypad return key should most probably remain an alias to the regular one.

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #1147

### CODE REVIEW

File: src/doomkeys.h [hunk 1]
```diff
@@ -24,70 +24,70 @@
 // This is the stuff configured by Setup.Exe.
 // Most key data are simple ascii (uppercased).
 //
-#define KEY_RIGHTARROW  0xae
```
Comment (fabiangreffrath) [inline src/doomkeys.h:27]: Now that we have clear text key names in the config file, we do not have to adhere to the original DOS scan codes anymore (i.e. we don't store their numerical values anyway). If we did, we'd end up with ambiguities e.g. with the keypad keys or other keys which send multi-byte scan codes.

File: src/m_input.c [hunk 2]
```diff
@@ -304,7 +304,22 @@ static const struct
   { KEY_INS,        "insert" },
   { KEY_PAUSE,      "pause" },
   { KEY_DEL,        "del" },
-  { KEY_PRTSCR,     "prtscr" }
+  { KEY_PRTSCR,     "prtscr" },
+  { KEYP_0,         "num0" },
```
Comment (fabiangreffrath) [inline src/m_input.c:308]: Or `kp0`? Or `keyp0`? Or `keypad0`?

Comment (rfomin) [reply src/m_input.c:308]: I like the current `KEYP_*` scheme.

**Review by rfomin — APPROVED** [review]
Woks good for me, thanks!

File: src/m_input.c (hunk 2 above)
Comment (fabiangreffrath) [reply src/m_input.c:308]: I was asking about the human readable string. 😉

Comment (rfomin) [reply src/m_input.c:308]: I think `num*` is good enough. I doubt that many users edit the config manually.

Comment (fabiangreffrath) [reply src/m_input.c:308]: Alright. I was unsure, because this is also the string presented in the "key bindings" menu.

Comment (rfomin) [reply src/m_input.c:308]: Just tested Quake, GZDoom and Eternity - no one supports num keys, interesting... Anyway, I think `num*` is good for menus too.

**Comment by fabiangreffrath** [comment]
Thank you!

**Pull request merged**